administrator/components/com_associations/models/associations.php

```diff
@@ -0,0 +1,298 @@
+				
+				if ($componentSplit[0] == 'com_menus')
+				{
+					$table = '#__menu';
+				}
+				else 
+				{
+					$table = str_replace("'", "", substr($componentModel, $start, $end - $start));
+				}
+				
+				$columns = $db->getTableColumns($table);
+
+				if (!isset($columns['title']))
+				{
+					$title = 'a.name';
+				}
+				else
+				{
+					$title = 'a.title';
+				}
+
+				$query->select('a.id, ' . $title . ' AS title, a.language, a.ordering');
+				
+				$query->from($db->quoteName($table, 'a'));
+
+				// Join over the language
+				$query->select('l.title AS language_title, l.image AS language_image')
+					->join('LEFT', $db->quoteName('#__languages') . ' AS l ON l.lang_code = a.language');
+
+				// Join over the associations.
+
+				if ($assoc)
+				{
+					$query->select('COUNT(' . $db->quoteName('asso2.id') . ') > 1 as ' . $db->quoteName('association'))
+						->join(
+							'LEFT',
+							$db->quoteName('#__associations', 'asso') . ' ON ' . $db->quoteName('asso.id') . ' = ' . $db->quoteName('a.id')
+							. ' AND ' . $db->quoteName('asso.context') . ' = ' . $db->quote($componentSplit[0] . '.item')
+						)
+						->join(
+							'LEFT',
+							$db->quoteName('#__associations', 'asso2') . ' ON ' . $db->quoteName('asso2.key') . ' = ' . $db->quoteName('asso.key')
+						)
+						->group(
+							$db->quoteName(
+								array(
+									'a.id',
+									$title,
+									'a.language'
+								)
+							)
+						);
+				}
+
+				if ($componentSplit[0] == 'com_menus')
+				{
+					// Exclude the root category.
+					$query->where('a.id > 1')
+						->where('a.client_id = 0');
+				}
+
+				// Filter on the language.
+				if ($language = $this->getState('associationlanguage'))
+				{
+					$query->where($db->quoteName('a.language') . ' = ' . $db->quote($language));
+				}
+
+				// Filter by search in name.
+				$search = $this->getState('filter.search');
+				if (!empty($search))
+				{
+					if (stripos($search, 'id:') === 0)
+					{
+						$query->where('a.id = ' . (int) substr($search, 3));
+					}
+					else
+					{
+						$search = $db->quote('%' . str_replace(' ', '%', $db->escape(trim($search), true) . '%'));
+						$query->where(
+							'(' . $db->quoteName($title) . ' LIKE ' . $search . ')'
+						);
+					}
+				}
+
+				// Add the list ordering clause.
+				$orderCol = $this->state->get('list.ordering', 'a.title');
+				$orderDirn = $this->state->get('list.direction', 'asc');
+
+				if ($orderCol == 'a.ordering')
+				{
+					$orderCol = 'title' . $orderDirn . ', a.ordering';
+				}
+
+				// SQL server change
+				if ($orderCol == 'language')
+				{
+					$orderCol = 'l.title';
+				}
+
+				$query->order($db->escape($orderCol . ' ' . $orderDirn));
+			}			
+		}
+		// If it's a category
+		elseif (strpos($component, '|'))
+		{
+			$componentSplit = explode('|', $component);
+			$extension = $componentSplit[1];
+
+			// Select the required fields from the table.
+			$query->select('a.id, a.title, a.language');
```

Comment:
don't know if it's needed here, but for consistency should be `a.title AS title`, or even better
`->select($db->quoteName('a.title', 'title');`

Note: a best practive is always quote db fields/table names because of different database system reserved names.